internal/graph/check.go

```diff
@@ -430,10 +446,20 @@ func mapFoundResources(result CheckResult, resourceType *core.RelationReference,
 func (cc *ConcurrentChecker) checkUsersetRewrite(ctx context.Context, crc currentRequestContext, rewrite *core.UsersetRewrite) CheckResult {
 	switch rw := rewrite.RewriteOperation.(type) {
 	case *core.UsersetRewrite_Union:
-		return union(ctx, crc, rw.Union.Child, cc.runSetOperation, cc.concurrencyLimit)
+		innerCtx := ctx
```

Comment:
not sure you need this, it seems fine to shadow like you do in the other branches

Reply:
you are right, I messed up while trying to only trace with there is more than one child